src/util/pointer_expr.h

```diff
@@ -1059,4 +1059,113 @@ inline void validate_expr(const object_size_exprt &value)
     "Object size expression must have pointer typed operand.");
 }
 
+/// A predicate that indicates that the object pointed to is live
+class live_object_exprt : public unary_predicate_exprt
```

Comment:
Should there be a check that the argument passed to the constructor is of pointer type?

Reply:
Sure, will add.